### DISCUSSION TITLE
Share static props with layout

### DISCUSSION THREAD

**Discussion by MauricioRobayo** [vercel/next.js#29329] [Help]
I'm fetching some data with `getStaticProps` that I want to make available to a layout, I'm not sure how to do this so I'm passing `pageProps` on `_app` to the `getLayout` function. 

Following the [With TypeScript](https://nextjs.org/docs/basic-features/layouts#with-typescript) example from the docs, It would look something like this:

```ts
import type { ReactElement, ReactNode } from 'react'
import type { NextPage } from 'next'
import type { AppProps } from 'next/app'

type NextPageWithLayout = NextPage & {
  getLayout?: (
    page: ReactElement,
    pageProps: SomePropsType // 👈 Added this
  ) => ReactNode;
};

type AppPropsWithLayout = AppProps & {
  Component: NextPageWithLayout
}

export default function MyApp({ Component, pageProps }: AppPropsWithLayout) {
  // Use the layout defined at the page level, if available
  const getLayout = Component.getLayout ?? ((page) => page)

  return getLayout(
    <Component {...pageProps} />,
    pageProps // 👈 Added this
  )
}
```

And on the pages sharing the layout I'm using something like this:

```ts
export const getStaticProps: GetStaticProps = async () => {
  const data = await getData();

  return {
    props: {
      data,
    },
  };
};

SomePage.getLayout = function getLayout(
  page: ReactElement,
  props: SomePropsType
) {
  return <PageLayout data={props.data}>{page}</PageLayout>;
};
```

This works so far, but I'm wondering if this is a good approach. I could not find a way of doing this in the documentation... any drawbacks? any issues about this approach that I should consider? Is there a better approach?

**Comment by abathellier** [comment]
That's what I'd do too, but i don't know if it's the best approach. It could be nice to add a section in the doc.

Btw they do that for nextjs commerce : https://github.com/vercel/commerce/blob/main/pages/_app.tsx

**Comment by JosephScript** [comment]
That's also what I'm doing. I think for now it's a good approach actually.
